test/unit/test.playMiddleware.js

```diff
@@ -0,0 +1,150 @@
+import pm from '../../src/playMiddleware.js';
+
+QUnit.module('Play middleware: not supported unit tests', window.sharedModuleHooks({
+  // let oldVersion;
+  // let oldUse;
+  // let oldMiddleware;
+  // let oldIOS;
+  // let oldAndroid;
+
+  beforeEach: function() {
+    this.sandbox.oldMiddleware = videojs.middleware ? videojs.mergeOptions({}, videojs.middleware) : undefined;
+
+    this.sandbox.stub(videojs, 'VERSION').get(() => {
+      return '6.0.0';
+    });
+    this.sandbox.stub(videojs, 'use').get(() => {
+      return undefined;
+    });
+    // this.sandbox.stub(videojs, 'middleware').get(() => {
+    //   return { TERMINATOR: 'fake terminator' };
+    // });
+    this.sandbox.stub(videojs, 'browser').get(() => {
+      return {
+        IS_ANDROID: false,
+        IS_IOS: false
+      };
+    });
+
+    // const mockVideojs = {
+    //   VERSION: '6.0.0',
+    //   browser: {
+    //     IS_ANDROID: false,
+    //     IS_IOS: false
+    //   }
+    // };
+    // this.oldVjs = window.videojs;
+    // window.videojs = mockVideojs;
+  }
+}));
+
+QUnit.test('isMiddlewareMediatorSupported is false if old video.js version', function(assert) {
+  this.sandbox.stub(videojs, 'VERSION').get(() => {
+    return '5.0.0';
+  });
+  // TODO stub out middleware somehow
+  sinon.stub(videojs.middleware, undefined);
+  assert.equal(pm.isMiddlewareMediatorSupported(), false,
+    'old video.js does not support middleware mediators');
+});
+
+QUnit.test('isMiddlewareMediatorSupported is false if on mobile', function(assert) {
+  assert.test.testEnvironment.sandbox.stub(videojs, 'browser').get(() => {
+    return {
+      IS_ANDROID: true,
+      IS_IOS: false
+    };
+  });
+  assert.equal(pm.isMiddlewareMediatorSupported(), false,
+    'is not supported on Android');
+
+  assert.test.testEnvironment.sandbox.stub(videojs, 'browser').get(() => {
+    return {
+      IS_ANDROID: false,
+      IS_IOS: true
+    };
+  });
+  assert.equal(pm.isMiddlewareMediatorSupported(), false,
+    'is not supported on iOS');
+});
+
+QUnit.test('playMiddleware callPlay will not terminate if not supported', function(assert) {
+  const m = pm.playMiddleware(this.player);
+  this.sandbox.stub(pm, 'isMiddlewareMediatorSupported').returns(false);
+  this.player.ads._shouldBlockPlay = true;
+  assert.equal(m.callPlay(), undefined,
+    'callPlay should not return an object');
+  assert.notEqual(typeof m.callPlay(), 'object',
+    'callPlay should not return the terminator');
+});
+
+QUnit.module('Play middleware: supported unit tests', window.sharedModuleHooks({
+  beforeEach: function() {
+    this.sandbox.stub(videojs, 'browser').get(() => {
+      return {
+        IS_ANDROID: false,
+        IS_IOS: false
+      };
+    });
+  }
+}));
+
+QUnit.skip('isMiddlewareMediatorSupported is true if middleware mediators exist on desktop', function(assert) {
```

Comment:
Should these be skipped?

Reply:
These tests still need to be updated to mock out video.js completely